tests/settings/FunctionMetadataTest.py

```diff
@@ -106,3 +106,61 @@ def check_function_metadata_data_set(self,
         self.assertEqual(metadata.retval_desc, retval_desc)
         self.assertEqual(metadata.non_optional_params, non_optional_params)
         self.assertEqual(metadata.optional_params, optional_params)
+
+    def test_filter_parameters(self):
+        metadata = FunctionMetadata(
+            "A", "DESC", "RETURNS SOMETHING",
+            {"x": ("-1", None),
+             "y": ("-2", None),
+             "z": ("-3", None),
+             "w": ("-4", None)},
+            {"a": ("1", None, 1), "b": ("2", None, 2), "c": ("3", None, 3)},
+            {"y", })
```

Comment:
testing omit functionality is superfluous because this is an unrelated property that is handled at a lower level. It makes your testcase overly complicated.